Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportInputController.cpp

```diff
@@ -0,0 +1,185 @@
+                if (m_mouseInteraction.m_mouseButtons.m_mouseButtons & mouseButtonValue)
+                {
+                    eventType = MouseEvent::Up;
+                }
+            }
+        }
+        else if (auto keyboardModifier = GetKeyboardModifier(event.m_inputChannel); keyboardModifier != KeyboardModifier::None)
+        {
+            if (state == InputChannel::State::Began || state == InputChannel::State::Updated)
+            {
+                m_mouseInteraction.m_keyboardModifiers.m_keyModifiers |= static_cast<AZ::u32>(keyboardModifier);
+            }
+            else if (state == InputChannel::State::Ended)
+            {
+                m_mouseInteraction.m_keyboardModifiers.m_keyModifiers &= ~static_cast<AZ::u32>(keyboardModifier);
+            }
+        }
+        else if (event.m_inputChannel.GetInputChannelId() == AzFramework::InputDeviceMouse::Movement::Z)
+        {
+            if (state == InputChannel::State::Began || state == InputChannel::State::Updated)
+            {
+                eventType = MouseEvent::Wheel;
+                wheelDelta = event.m_inputChannel.GetValue();
+            }
+        }
+
+        if (eventType)
+        {
+            MouseInteraction mouseInteraction = m_mouseInteraction;
+            if (overrideButton)
+            {
+                mouseInteraction.m_mouseButtons.m_mouseButtons = static_cast<AZ::u32>(overrideButton.value());
+            }
+
+            mouseInteraction.m_interactionId.m_viewportId = GetViewportId();
+
+            auto currentCursorState = AzFramework::SystemCursorState::Unknown;
+            AzFramework::InputSystemCursorRequestBus::EventResult(
+                currentCursorState, event.m_inputChannel.GetInputDevice().GetInputDeviceId(),
+                &AzFramework::InputSystemCursorRequestBus::Events::GetSystemCursorState);
+
+            const auto mouseInteractionEvent = [mouseInteraction, event = eventType.value(), wheelDelta,
+                                                cursorCaptured = currentCursorState == AzFramework::SystemCursorState::ConstrainedAndHidden]
+            {
+                switch (event)
+                {
+                case MouseEvent::Up:
+                case MouseEvent::Down:
+                case MouseEvent::Move:
+                case MouseEvent::DoubleClick:
+                    return MouseInteractionEvent(AZStd::move(mouseInteraction), event, cursorCaptured);
+                case MouseEvent::Wheel:
+                    return MouseInteractionEvent(AZStd::move(mouseInteraction), wheelDelta);
+                }
+
+                AZ_Assert(false, "Unhandled MouseEvent");
+                return MouseInteractionEvent(MouseInteraction{}, MouseEvent::Up, false);
+            }();
+
+            ManipulatorRequestBus::BroadcastResult(interactionHandled, &ManipulatorRequests::HandleMouseEvent, mouseInteractionEvent);
```

Comment:
fix: I believe here we should use `EventResult` and use the new `ManipulatorManagerId`. We want to avoid `Broadcast` as this will then also go to the main viewport potentially? (unless this is a new EMFX bus but even so having it use a Id is likely a good idea).

Reply:
This one has a single address policy which doesn't need to use eventResult to deliver to specific address.